supervisor/utils/pwned.py

```diff
@@ -10,10 +10,15 @@
 _LOGGER: logging.Logger = logging.getLogger(__name__)
 _API_CALL = "https://api.pwnedpasswords.com/range/{hash}"
 
+_CACHE = set()
+
 
 async def check_pwned_password(websession: aiohttp.ClientSession, sha1_pw: str) -> bool:
     """Check if password is pwned."""
     sha1_pw = sha1_pw.upper()
+    if sha1_pw[:5] in _CACHE:
```

Comment:
```suggestion
    sha1_pw = sha1_pw.upper()[:5]
    if sha1_pw in _CACHE:
```
You are now stripping the string 3 times, that is not needed